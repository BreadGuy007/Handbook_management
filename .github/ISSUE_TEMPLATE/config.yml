blank_issues_enabled: false
contact_links:
  - name: General help request
    url: https://wordpress.org/support/forum/how-to-and-troubleshooting/
    about: For general help requests, create a new topic in the Fixing WordPress support forum
  - name: Technical help request
    url: https://wordpress.org/support/forum/wp-advanced/
    about: For more technical help requests, create a new topic in the Developing with WordPress Forum
  - name: Development help request
    url: https://wordpress.stackexchange.com/
    about: For questions about WordPress development, ask a question in the WordPress Development Stack Exchange
