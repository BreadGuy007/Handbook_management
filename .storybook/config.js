import { configure } from '@storybook/react';
import * as element from 'element';
import './style.scss';

function loadStories() {
	window.wp = { ...window.wp, element };
	require( '../components/story' );
	require( '../components/button/story' );
}

configure( loadStories, module );
