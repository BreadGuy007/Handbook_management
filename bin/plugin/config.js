const gitRepoOwner = 'WordPress';

const config = {
	slug: 'gutenberg',
	name: 'Gutenberg',
	team: 'Gutenberg Core',
	githubRepositoryOwner: gitRepoOwner,
	githubRepositoryName: 'gutenberg',
	pluginEntryPoint: 'gutenberg.php',
	buildZipCommand: '/bin/bash bin/build-plugin-zip.sh',
	wpRepositoryReleasesURL:
		'https://github.com/WordPress/gutenberg/releases/tag/',
	gitRepositoryURL: 'https://github.com/' + gitRepoOwner + '/gutenberg.git',
	svnRepositoryURL: 'https://plugins.svn.wordpress.org/gutenberg',
};

module.exports = config;
