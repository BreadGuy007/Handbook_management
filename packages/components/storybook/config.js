/**
 * External dependencies
 */
import { addDecorator, configure } from '@storybook/react';
import { withA11y } from '@storybook/addon-a11y';

/**
 * Internal dependencies
 */
import '../build-style/style.css';

addDecorator( withA11y );
configure(
	[
		require.context( '../docs', true, /\/.+\.mdx$/ ),
		require.context( '../src', true, /\/stories\/.+\.js$/ ),
	],
	module
);
