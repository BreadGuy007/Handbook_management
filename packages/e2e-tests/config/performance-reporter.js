const { readFileSync, existsSync } = require( 'fs' );

function average( array ) {
	return array.reduce( ( a, b ) => a + b ) / array.length;
}

function round( number, decimalPlaces = 2 ) {
	const factor = Math.pow( 10, decimalPlaces );
	return Math.round( number * factor ) / factor;
}

class PerformanceReporter {
	onRunComplete() {
		const path = __dirname + '/../specs/performance/results.json';

		if ( ! existsSync( path ) ) {
			return;
		}

		const results = readFileSync( path, 'utf8' );
		const { load, domcontentloaded, type } = JSON.parse( results );

		// eslint-disable-next-line no-console
		console.log( `
Average time to load: ${ round( average( load ) ) }ms
Average time to DOM content load: ${ round( average( domcontentloaded ) ) }ms
Average time to type character: ${ round( average( type ) ) }ms
Slowest time to type character: ${ round( Math.max( ...type ) ) }ms
Fastest time to type character: ${ round( Math.min( ...type ) ) }ms
		` );
	}
}

module.exports = PerformanceReporter;
