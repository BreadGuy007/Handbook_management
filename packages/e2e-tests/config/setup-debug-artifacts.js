/**
 * External dependencies
 */
const fs = require( 'fs' );
const util = require( 'util' );
const root = process.env.GITHUB_WORKSPACE || process.cwd();
const ARTIFACTS_PATH = root + '/artifacts';

const writeFile = util.promisify( fs.writeFile );

if ( ! fs.existsSync( ARTIFACTS_PATH ) ) {
	fs.mkdirSync( ARTIFACTS_PATH );
}

/**
 * Gutenberg uses the default jest-jasmine2 test runner that comes with Jest.
 * Unfortunately, version 2 of jasmine doesn't support async reporters. It
 * does support async before and after hooks though,  so the workaround here
 * works by making each test wait for the artifacts before starting.
 *
 * Kudos to Tom Esterez (@testerez) for sharing this idea in https://github.com/smooth-code/jest-puppeteer/issues/131#issuecomment-424073620
 */
let artifactsPromise;
// eslint-disable-next-line jest/no-jasmine-globals
jasmine.getEnv().addReporter( {
	specDone: ( result ) => {
		if ( result.status === 'failed' ) {
			artifactsPromise = storeArtifacts( result.fullName );
		}
	},
} );

beforeEach( () => artifactsPromise );
afterAll( () => artifactsPromise );

async function storeArtifacts( testName ) {
	const slug = slugify( testName );
	await writeFile(
		`${ ARTIFACTS_PATH }/${ slug }-snapshot.html`,
		await page.content()
	);
	await page.screenshot( { path: `${ ARTIFACTS_PATH }/${ slug }.jpg` } );
}

function slugify( testName ) {
	const datetime = new Date().toISOString().split( '.' )[ 0 ];
	const readableName = `${ testName } ${ datetime }`;
	const slug = readableName
		.toLowerCase()
		.replace( /:/g, '-' )
		.replace( /[^0-9a-zA-Z \-\(\)]/g, '' )
		.replace( / /g, '-' );
	return slug;
}
