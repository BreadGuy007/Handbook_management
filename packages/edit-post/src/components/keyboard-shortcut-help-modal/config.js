/**
 * WordPress dependencies
 */
import { displayShortcutList, shortcutAriaLabel } from '@wordpress/keycodes';
import { __ } from '@wordpress/i18n';

const {
	// Cmd+<key> on a mac, Ctrl+<key> elsewhere.
	primary,
	// Shift+Cmd+<key> on a mac, Ctrl+Shift+<key> elsewhere.
	primaryShift,
	// Ctrl+Alt+<key> on a mac, Shift+Alt+<key> elsewhere.
	access,
	ctrl,
	alt,
	ctrlShift,
} = displayShortcutList;

export const mainShortcuts = [
	{
		keyCombination: access( 'h' ),
		description: __( 'Display these keyboard shortcuts.' ),
	},
];

export const globalShortcuts = [
	'core/editor/save',
	'core/editor/undo',
	'core/editor/redo',
	'core/edit-post/toggle-sidebar',
	{
		keyCombination: access( 'o' ),
		description: __( 'Open the block navigation menu.' ),
	},
	{
		keyCombination: ctrl( '`' ),
		description: __( 'Navigate to the next part of the editor.' ),
		ariaLabel: shortcutAriaLabel.ctrl( '`' ),
	},
	{
		keyCombination: ctrlShift( '`' ),
		description: __( 'Navigate to the previous part of the editor.' ),
		ariaLabel: shortcutAriaLabel.ctrlShift( '`' ),
	},
	{
		keyCombination: access( 'n' ),
		description: __( 'Navigate to the next part of the editor (alternative).' ),
	},
	{
		keyCombination: access( 'p' ),
		description: __( 'Navigate to the previous part of the editor (alternative).' ),
	},
	{
		keyCombination: alt( 'F10' ),
		description: __( 'Navigate to the nearest toolbar.' ),
	},
	'core/edit-post/toggle-mode',
];

export const textFormattingShortcuts = [
	{
		keyCombination: primary( 'b' ),
		description: __( 'Make the selected text bold.' ),
	},
	{
		keyCombination: primary( 'i' ),
		description: __( 'Make the selected text italic.' ),
	},
	{
		keyCombination: primary( 'k' ),
		description: __( 'Convert the selected text into a link.' ),
	},
	{
		keyCombination: primaryShift( 'k' ),
		description: __( 'Remove a link.' ),
	},
	{
		keyCombination: primary( 'u' ),
		description: __( 'Underline the selected text.' ),
	},
];
