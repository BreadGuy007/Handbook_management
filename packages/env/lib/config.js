'use strict';
/**
 * External dependencies
 */
const fs = require( 'fs' ).promises;
const path = require( 'path' );
const os = require( 'os' );
const crypto = require( 'crypto' );

/**
 * Internal dependencies
 */
const detectDirectoryType = require( './detect-directory-type' );

/**
 * Error subtype which indicates that an expected validation erorr occured
 * while reading wp-env configuration.
 */
class ValidationError extends Error {}

/**
 * The string at the beginning of a source path that points to a home-relative
 * directory. Will be '~/' on unix environments and '~\' on Windows.
 */
const HOME_PATH_PREFIX = `~${ path.sep }`;

/**
 * A wp-env config object.
 *
 * @typedef Config
 * @property {string} name Name of the environment.
 * @property {string} configDirectoryPath Path to the .wp-env.json file.
 * @property {string} workDirectoryPath Path to the work directory located in ~/.wp-env.
 * @property {string} dockerComposeConfigPath Path to the docker-compose.yml file.
 * @property {Source|null} coreSource The WordPress installation to load in the environment.
 * @property {Source[]} pluginSources Plugins to load in the environment.
 * @property {Source[]} themeSources Themes to load in the environment.
 * @property {number} port The port on which to start the development WordPress environment.
 * @property {number} testsPort The port on which to start the testing WordPress environment.
 */

/**
 * A WordPress installation, plugin or theme to be loaded into the environment.
 *
 * @typedef Source
 * @property {string} type The source type. Can be 'local' or 'git'.
 * @property {string} path The path to the WordPress installation, plugin or theme.
 * @property {string} basename Name that identifies the WordPress installation, plugin or theme.
 */

module.exports = {
	ValidationError,

	/**
	 * Reads and parses the given .wp-env.json file into a wp-env config object.
	 *
	 * @param {string} configPath Path to the .wp-env.json file.
	 * @return {Config} A wp-env config object.
	 */
	async readConfig( configPath ) {
		const configDirectoryPath = path.dirname( configPath );

		let config = null;

		try {
			config = JSON.parse( await fs.readFile( configPath, 'utf8' ) );
		} catch ( error ) {
			if ( error.code === 'ENOENT' ) {
				// Config file does not exist. Do nothing - it's optional.
			} else if ( error instanceof SyntaxError ) {
				throw new ValidationError(
					`Invalid .wp-env.json: ${ error.message }`
				);
			} else {
				throw new ValidationError(
					`Could not read .wp-env.json: ${ error.message }`
				);
			}
		}

		if ( config === null ) {
			const type = await detectDirectoryType( configDirectoryPath );
			if ( type === 'core' ) {
				config = { core: '.' };
			} else if ( type === 'plugin' ) {
				config = { plugins: [ '.' ] };
			} else if ( type === 'theme' ) {
				config = { themes: [ '.' ] };
			} else {
				throw new ValidationError(
					`No .wp-env.json file found at '${ configPath }' and could not determine if '${ configDirectoryPath }' is a WordPress installation, a plugin, or a theme.`
				);
			}
		}

		config = Object.assign(
			{
				core: null,
				plugins: [],
				themes: [],
				port: 8888,
				testsPort: 8889,
			},
			config
		);

		config.port = getNumberFromEnvVariable( 'WP_ENV_PORT' ) || config.port;
		config.testsPort =
			getNumberFromEnvVariable( 'WP_ENV_TESTS_PORT' ) || config.testsPort;

		if ( config.core !== null && typeof config.core !== 'string' ) {
			throw new ValidationError(
				'Invalid .wp-env.json: "core" must be null or a string.'
			);
		}

		if (
			! Array.isArray( config.plugins ) ||
			config.plugins.some( ( plugin ) => typeof plugin !== 'string' )
		) {
			throw new ValidationError(
				'Invalid .wp-env.json: "plugins" must be an array of strings.'
			);
		}

		if (
			! Array.isArray( config.themes ) ||
			config.themes.some( ( theme ) => typeof theme !== 'string' )
		) {
			throw new ValidationError(
				'Invalid .wp-env.json: "themes" must be an array of strings.'
			);
		}

		if ( ! Number.isInteger( config.port ) ) {
			throw new ValidationError(
				'Invalid .wp-env.json: "port" must be an integer.'
			);
		}

		if ( ! Number.isInteger( config.testsPort ) ) {
			throw new ValidationError(
				'Invalid .wp-env.json: "testsPort" must be an integer.'
			);
		}

		if ( config.port === config.testsPort ) {
			throw new ValidationError(
				'Invalid .wp-env.json: "testsPort" and "port" must be different.'
			);
		}

		const workDirectoryPath = path.resolve(
			os.homedir(),
			'.wp-env',
			md5( configPath )
		);

		return {
			name: path.basename( configDirectoryPath ),
			configDirectoryPath,
			workDirectoryPath,
			port: config.port,
			testsPort: config.testsPort,
			dockerComposeConfigPath: path.resolve(
				workDirectoryPath,
				'docker-compose.yml'
			),
			coreSource: includeTestsPath(
				parseSourceString( config.core, {
					workDirectoryPath,
				} )
			),
			pluginSources: config.plugins.map( ( sourceString ) =>
				parseSourceString( sourceString, {
					workDirectoryPath,
				} )
			),
			themeSources: config.themes.map( ( sourceString ) =>
				parseSourceString( sourceString, {
					workDirectoryPath,
				} )
			),
		};
	},
};

/**
 * Parses a source string into a source object.
 *
 * @param {string|null} sourceString The source string. See README.md for documentation on valid source string patterns.
 * @param {Object} options
 * @param {boolean} options.hasTests Whether or not a `testsPath` is required. Only the 'core' source needs this.
 * @param {string} options.workDirectoryPath Path to the work directory located in ~/.wp-env.
 * @return {Source|null} A source object.
 */
function parseSourceString( sourceString, { workDirectoryPath } ) {
	if ( sourceString === null ) {
		return null;
	}

	if (
		sourceString.startsWith( '.' ) ||
		sourceString.startsWith( HOME_PATH_PREFIX ) ||
		path.isAbsolute( sourceString )
	) {
		let sourcePath;
		if ( sourceString.startsWith( HOME_PATH_PREFIX ) ) {
			sourcePath = path.resolve(
				os.homedir(),
				sourceString.slice( HOME_PATH_PREFIX.length )
			);
		} else {
			sourcePath = path.resolve( sourceString );
		}
		const basename = path.basename( sourcePath );
		return {
			type: 'local',
			path: sourcePath,
			basename,
		};
	}

	const gitHubFields = sourceString.match( /^([^\/]+)\/([^#]+)(?:#(.+))?$/ );
	if ( gitHubFields ) {
		return {
			type: 'git',
			url: `https://github.com/${ gitHubFields[ 1 ] }/${ gitHubFields[ 2 ] }.git`,
			ref: gitHubFields[ 3 ] || 'master',
			path: path.resolve( workDirectoryPath, gitHubFields[ 2 ] ),
			basename: gitHubFields[ 2 ],
		};
	}

	throw new ValidationError(
		`Invalid or unrecognized source: "${ sourceString }."`
	);
}

/**
 * Given a source object, returns a new source object with the testsPath
 * property set correctly. Only the 'core' source requires a testsPath.
 *
 * @param {Source|null} source A source object.
 * @return {Source|null} A source object.
 */
function includeTestsPath( source ) {
	if ( source === null ) {
		return null;
	}

	return {
		...source,
		testsPath: path.resolve(
			source.path,
			'..',
			'tests-' + path.basename( source.path )
		),
	};
}

/**
 * Parses an environment variable which should be a number.
 *
 * Throws an error if the variable cannot be parsed to a number.
 * Returns null if the environment variable has not been specified.
 *
 * @param {string} varName The environment variable to check (e.g. WP_ENV_PORT).
 * @return {null|number} The number. Null if it does not exist.
 */
function getNumberFromEnvVariable( varName ) {
	// Allow use of the default if it does not exist.
	if ( ! process.env[ varName ] ) {
		return null;
	}

	const maybeNumber = parseInt( process.env[ varName ] );

	// Throw an error if it is not parseable as a number.
	if ( isNaN( maybeNumber ) ) {
		throw new ValidationError(
			`Invalid environment variable: ${ varName } must be a number.`
		);
	}

	return maybeNumber;
}

/**
 * Hashes the given string using the MD5 algorithm.
 *
 * @param {string} data The string to hash.
 * @return {string} An MD5 hash string.
 */
function md5( data ) {
	return crypto
		.createHash( 'md5' )
		.update( data )
		.digest( 'hex' );
}
