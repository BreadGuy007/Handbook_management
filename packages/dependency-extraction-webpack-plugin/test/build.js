/**
 * External dependencies
 */
const fs = require( 'fs' );
const glob = require( 'glob' ).sync;
const mkdirp = require( 'mkdirp' ).sync;
const path = require( 'path' );
const rimraf = require( 'rimraf' ).sync;
const webpack = require( 'webpack' );

const fixturesPath = path.join( __dirname, 'fixtures' );
const configFixtures = fs.readdirSync( fixturesPath ).sort();

afterAll( () => rimraf( path.join( __dirname, 'build' ) ) );

describe.each( configFixtures )( 'Webpack `%s`', ( configCase ) => {
	const testDirectory = path.join( fixturesPath, configCase );
	const outputDirectory = path.join( __dirname, 'build', configCase );

	beforeEach( () => {
		rimraf( outputDirectory );
		mkdirp( outputDirectory );
	} );

	// This afterEach is necessary to prevent watched tests from retriggering on every run.
	afterEach( () => rimraf( outputDirectory ) );

	test( 'should produce expected output', () =>
		new Promise( ( resolve ) => {
			const options = Object.assign(
				{
					context: testDirectory,
					entry: './index.js',
					mode: 'production',
					optimization: {
						minimize: false,
						chunkIds: 'named',
						moduleIds: 'named',
					},
					output: {},
				},
				require( path.join( testDirectory, 'webpack.config.js' ) )
			);
			options.output.path = outputDirectory;

			webpack( options, ( err, stats ) => {
				expect( err ).toBeNull();

				const assetFiles = glob(
					`${ outputDirectory }/+(*.asset|assets).@(json|php)`
				);
				const hasCombinedAssets = ( options.plugins || [] ).some(
					( plugin ) => !! ( plugin.options || {} ).combineAssets
				);
				const entrypointCount =
					typeof options.entry === 'object'
						? Object.keys( options.entry ).length
						: 1;
				const expectedLength = hasCombinedAssets ? 1 : entrypointCount;
				expect( assetFiles ).toHaveLength( expectedLength );

				// Asset files should match.
				assetFiles.forEach( ( assetFile ) => {
					expect(
						fs.readFileSync( assetFile, 'utf-8' )
					).toMatchSnapshot( 'Asset file should match snapshot' );
				} );

				// Webpack stats external modules should match.
				const externalModules = stats.compilation.modules
					.filter( ( { external } ) => external )
					.sort()
					.map( ( module ) => ( {
						externalType: module.externalType,
						request: module.request,
						userRequest: module.userRequest,
					} ) );
				expect( externalModules ).toMatchSnapshot(
					'External modules should match snapshot'
				);

				resolve();
			} );
		} ) );
} );
