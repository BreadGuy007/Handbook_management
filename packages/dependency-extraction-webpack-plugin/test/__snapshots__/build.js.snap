// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack \`dynamic-import\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '5d8e58fe98bc4c6277a76ece11fcb8b7');"`;

exports[`Webpack \`dynamic-import\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "blob",
      ],
    },
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": "lodash",
    },
    "userRequest": "lodash",
  },
]
`;

exports[`Webpack \`function-output-filename\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '407eacfb62231e58aaca5eec87e63725');"`;

exports[`Webpack \`function-output-filename\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "blob",
      ],
    },
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": "lodash",
    },
    "userRequest": "lodash",
  },
]
`;

exports[`Webpack \`no-default\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array(), 'version' => '6e00af916c16fcb88817e1c13ec5f754');"`;

exports[`Webpack \`no-default\` should produce expected output: External modules should match snapshot 1`] = `Array []`;

exports[`Webpack \`no-deps\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array(), 'version' => 'e8e65570823ee9206bdada3a0f11b610');"`;

exports[`Webpack \`no-deps\` should produce expected output: External modules should match snapshot 1`] = `Array []`;

exports[`Webpack \`output-format-json\` should produce expected output: Asset file should match snapshot 1`] = `"{\\"dependencies\\":[\\"lodash\\"],\\"version\\":\\"0e6d34ea09104a64e6184f524b48ad65\\"}"`;

exports[`Webpack \`output-format-json\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "this",
    "request": Object {
      "this": "lodash",
    },
    "userRequest": "lodash",
  },
]
`;

exports[`Webpack \`overrides\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('wp-blob', 'wp-script-handle-for-rxjs', 'wp-url'), 'version' => '2fd24c89b50c763f2e2dcc02a6933c16');"`;

exports[`Webpack \`overrides\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "blob",
      ],
    },
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "url",
      ],
    },
    "userRequest": "@wordpress/url",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "rxjs",
        "operators",
      ],
    },
    "userRequest": "rxjs/operators",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": "rxjs",
    },
    "userRequest": "rxjs",
  },
]
`;

exports[`Webpack \`with-externs\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('wp-url'), 'version' => 'e987036543f42978608ac27e589b5fe3');"`;

exports[`Webpack \`with-externs\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "var",
    "request": "wp.blob",
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "url",
      ],
    },
    "userRequest": "@wordpress/url",
  },
  Object {
    "externalType": "var",
    "request": "rxjs.operators",
    "userRequest": "rxjs/operators",
  },
  Object {
    "externalType": "var",
    "request": "rxjs",
    "userRequest": "rxjs",
  },
]
`;

exports[`Webpack \`wordpress\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => 'f6f0d227edbcb02382070c0efac33b51');"`;

exports[`Webpack \`wordpress\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "blob",
      ],
    },
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": "lodash",
    },
    "userRequest": "lodash",
  },
]
`;

exports[`Webpack \`wordpress-require\` should produce expected output: Asset file should match snapshot 1`] = `"<?php return array('dependencies' => array('lodash', 'wp-blob'), 'version' => '31a7631b4838830b14b5ab10e17f3e0f');"`;

exports[`Webpack \`wordpress-require\` should produce expected output: External modules should match snapshot 1`] = `
Array [
  Object {
    "externalType": "this",
    "request": Object {
      "this": Array [
        "wp",
        "blob",
      ],
    },
    "userRequest": "@wordpress/blob",
  },
  Object {
    "externalType": "this",
    "request": Object {
      "this": "lodash",
    },
    "userRequest": "lodash",
  },
]
`;
